Source/Venice/Channel/Channel.swift

```diff
@@ -67,26 +67,23 @@ public final class Channel<T>: Sequence {
     }
 
     /// Closes the channel. When a channel is closed it cannot receive values anymore.
-    public func close() -> Bool {
-        if closed {
-            return false
-        }
+    public func close() throws {
+        guard !closed else { return }
```

Comment:
That's the right way to do it. 
If trying to close and already closed that's not an error but a convenient no op. 

Sent from my iPhone

> On Apr 15, 2016, at 9:42 PM, Paulo Faria <notifications@github.com> wrote:
> 
> In Source/Venice/Channel/Channel.swift:
> 
> > @@ -67,26 +67,23 @@ public final class Channel<T>: Sequence {
> >      }
> >  
> >      /// Closes the channel. When a channel is closed it cannot receive values anymore.
> > -    public func close() -> Bool {
> > -        if closed {
> > -            return false
> > -        }
> > +    public func close() throws {
> > +        guard !closed else { return }
> I'm not sure it should have similar API. Specially if it's not adopting Closable. The only way this close can fail is is if the channel is already closed. And we could even argue, would this be considered an error? Or are we using the return value just as a convenience? Seems to me in this case would be better to make it public func close() and if the consumer wants to know if the channel is already closed he/she could just check the closed variable.
> 
> ―
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>